man/sysupdate.d.xml

```diff
@@ -0,0 +1,872 @@
+    defined in. First, the resource data is downloaded directly into a target file/directory/partition. Once
+    this is completed for all defined transfers, in a second step the files/directories/partitions are
+    renamed to their final names as defined by the target <varname>MatchPattern=</varname>, again in the
+    order the <filename>.conf</filename> transfer file names dictate. This step is not atomic, however it is
+    guaranteed to be executed strictly in order with suitable disk synchronization in place. Typically, when
+    updating an OS one of the transfers defines the entry point when booting. Thus it is generally a good idea
+    to order the resources via the transfer configuration file names so that the entry point is finalized
+    last, ensuring that any abnormal termination does not leave an entry point around whose backing is not
+    established yet. In the example above it would hence make sense to establish the EFI kernel image last
+    and thus give its transfer configuration file the alphabetically last name.</para>
+
+    <para>See below for an extended, more specific example based on the above.</para>
+  </refsect1>
+
+  <refsect1>
+    <title>Resource Types</title>
+
+    <para>Each transfer file defines one source resource to transfer to one target resource. The following
+    resource types are supported:</para>
+
+    <orderedlist>
+
+      <listitem><para>Resources of type <literal>http-file</literal> encapsulate a file on a web server,
+      referenced via a HTTP or HTTPS URL. When an update takes place, the file is downloaded and decompressed
+      and then written to the target file or partition. This resource type is only available for sources, not
+      for targets. The list of available versions of resources of this type is encoded in
+      <filename>SHA256SUMS</filename> manifest files, accompanied by
+      <filename>SHA256SUMS.gpg</filename> detached signatures.</para></listitem>
+
+      <listitem><para>The <literal>http-tar</literal> resource type is similar, but the file must be a
+      <filename>.tar</filename> archive. When an update takes place, the file is decompressed and unpacked
+      into a directory or btrfs subvolume. This resource type is only available for sources, not for
+      targets. Just like <literal>http-file</literal>, <literal>http-tar</literal> version enumeration makes
+      use of <filename>SHA256SUMS</filename> files, authenticated via
+      <filename>SHA256SUMS.gpg</filename>.</para></listitem>
+
+      <listitem><para>The <literal>regular-file</literal> resource type encapsulates a local regular file on
+      disk. During updates the file is uncompressed and written to the target file or partition. This
+      resource type is available both as source and as target. When updating no integrity or authentication
+      verification is done for resources of this type.</para></listitem>
```

Comment:
Why not? Given it's already supported for other transports, it seems easy and useful to be able to do some checks before using these too?

Reply:
The idea is that it's the simplest logic, that is trivial to use because it requires almost no setup. Signing and getting keys to the consumers is not trivial though, hence I left it out.

More importantly though: you can actually use http-file resource type instead, and then specify a file:// instead of an http:// or https:// and get local file system behaviour. So the support actually *does* exist. I didn't push people towards this so far far though since using something called "http" and then specifying "file:" URLs with it is quite confusing...

Reply:
wouldn't it be useful if it wasn't mandatory, but if the hashsum file was found it was automatically used?

Reply:
So the current behaviour is this:

transport regular-file: local trusted sources only; no checksumming, no signing, and not even opt-in for it.

transport http-file: payloads comes via potentially untrusted transports thus insist on checksumming and signing (but do allow opt-out of signing and optionally also from checksumming). Also, allow `file://` URLs in case you want to check signatures/checksumming even on local, trusted sources.

I mean, the checksumming/signing is really there to establish trust in a cryptographic sense, not as a method to detect IO issues where bits are swapped. Thus when doing local transfer it's not the right tool (or to say it differently: a CRC check would probably be a better idea for that). By not doing any of that for the local transfers we have the ability to use do reflinks/copy_file_range() and all that stuff, because we don#t have to actually to read anything.

or to say this in a different way: if you want checksumming/signing, then use http-file with a file: URL and you should be good. And there doing checksumming/signing is opt-out, so you have to go through extra hoops to turn off the verifications.

Reply:
Does using `file://` require spawning a local http process to serve the files? Or does curl figure out it just needs to access the local filesystem?

Reply:
> Does using `file://` require spawning a local http process to serve the files? Or does curl figure out it just needs to access the local filesystem?

No, not http daemon involved in that case. Just curl going directly to disk.

Reply:
ok, looks good to me then